scripts/restore-sanitised-database.sh

```diff
@@ -63,9 +78,29 @@ case $ACCOUNT in
         ;;
 esac
 
+
+# Default values for variables
+DEFAULT_SANITISED_DB_NAME="sanitise-db"
+
 # Variables based on the environment
-SANITISED_DB="Sanitised-database.sql"
-S3_BUCKET="aws-s3-bucket-sanitise-database"
+SANITISED_DB_NAME="${1:-$DEFAULT_SANITISED_DB_NAME}"
+S3_BUCKET=""
+
+
+
+# List all S3 buckets and get their names
+bucket_names=$(aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- aws s3api list-buckets --query "Buckets[].Name" --output json)
+
+for bucket in $(echo "$bucket_names" | jq -r '.[]'); do
+    TARGET_BUCKET=$(aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- aws s3api get-bucket-tagging --bucket "$bucket")
+    SANITISE_BUCKET=$(echo "$TARGET_BUCKET" | jq -r '.TagSet[] | select(.Key == "sanitise" and .Value == "db")')
+
+    if [ -n "$SANITISE_BUCKET" ];
+    then
+        S3_BUCKET="$bucket"
+        break
+    fi
```

Comment:
nice :) we'll obvs need to change the tags you're filtering on before merging this (once Gideon's PR creating the bucket is merged), but this looks sound in principle :)